live_pipeline/schema.sql

```diff
@@ -0,0 +1,98 @@
+-- cleanup
+
+DROP TABLE IF EXISTS review;
+DROP TABLE IF EXISTS genre;
+DROP TABLE IF EXISTS game;
+DROP TABLE IF EXISTS developer;
+DROP TABLE IF EXISTS publisher;
+DROP TABLE IF EXISTS platform;
+
+-- tables with no foreign key
+
+CREATE TABLE developer(
+    developer_id INT GENERATED ALWAYS AS IDENTITY,
+    developer_name TEXT NOT NULL,
+    PRIMARY KEY (developer_id)
+
+);
+
+CREATE TABLE publisher(
+    publisher_id INT GENERATED ALWAYS AS IDENTITY,
+    publisher_name TEXT NOT NULL,
+    PRIMARY KEY (publisher_id)
+
+);
+
+CREATE TABLE platform(
+    platform_id INT GENERATED ALWAYS AS IDENTITY,
+    mac BOOLEAN NOT NULL,
+    windows BOOLEAN NOT NULL,
+    linux BOOLEAN NOT NULL,
+    PRIMARY KEY (platform_id)
+
+);
+
+-- game references the three above tables
+
+
+CREATE TABLE game(
+    game_id INT GENERATED ALWAYS AS IDENTITY,
+    app_id INT NOT NULL,
+    title TEXT NOT NULL,
+    release_date DATE NOT NULL,
+    price FLOAT NOT NULL,
+    sale_price FLOAT NOT NULL,
+    developer_id INT,
+    publisher_id INT,
+    platform_id INT,
+    PRIMARY KEY (game_id),
+    FOREIGN KEY (developer_id) REFERENCES developer(developer_id),
+    FOREIGN KEY (publisher_id) REFERENCES publisher(publisher_id),
+    FOREIGN KEY (platform_id) REFERENCES platform(platform_id)
+
+);
+
+
+-- review and genre reference game
+
+-- default 0 might make sentiment analysis loading easier, but it's easy to take out if we don't use it
+
+CREATE TABLE review(
+    review_id INT GENERATED ALWAYS AS IDENTITY,
+    sentiment FLOAT NOT NULL DEFAULT 0,
+    review_text TEXT, 
```

Comment:
Is there a reason for not having a `NOT NULL` constraint? 

- Do we want to add reviews with missing text?
- Are there reviews with missing text?

Reply:
I think this should get dropped in transform but yes agree would be wise to put in here as a double security measure